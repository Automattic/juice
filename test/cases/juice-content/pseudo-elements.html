<html><body>
<style>
body {
  --var: 'Element';
  counter-reset: element 20 other;
}

a {
  text-decoration: underline;
}

a:before,
a:after {
  content: "a";
}

* a:before {
  content: 'Â®\+';
}

a:after,
a::after {
  font-weight: bold;
}

a:first-child {
  color: blue;
}

b:after {
 content: " ";
}

b:after {
 content: "b";
 color: green;
}

b :last-child {
  color: red;
}

c:not(.not) {
    color: red;
}

d:before {
  content: url(data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs=);
}

d:after {
  content: url('data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs=');
  width: 20%;
}

e {
  counter-increment: element other -2;
}

e:before {
  content: var(--var) " / " counter(element) '. ' counter(other);
}

e[length]:after {
  content: ' | Length: ' attr(length) ' [' counter(element, lower-roman) ':' counter(element, upper-latin) ']';
}

f {
  counter-increment: other;
}

f:before {
  content: counter(other);
}

</style>
<a href="#">Test</a>
<b></b>
<b><span></span></b>
<c></c>
<c class="not"></c>
<d><span>test</span></d>
<e><span></span></e>
<e length="100"><span>Test</span></e>
<f><span></span></f>
<f><span></span></f>
</body></html>
