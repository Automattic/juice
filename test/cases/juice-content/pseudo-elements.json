{
  "url": "./",
  "removeStyleTags": true,
  "inlinePseudoElements": true,
  "resolveCSSVariables": false
}
