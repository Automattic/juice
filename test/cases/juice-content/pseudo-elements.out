<html><body style="--var: 'Element'; counter-reset: element 20 other;">

<a href="#" style="text-decoration: underline; color: blue;"><span>Â®+</span>Test<span style="font-weight: bold;">a</span></a>
<b><span style="color: green;">b</span></b>
<b><span style="color: red;"></span><span style="color: green;">b</span></b>
<c style="color: red;"></c>
<c class="not"></c>
<d><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs="><span>test</span><img style="width: 20%;" src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs="></d>
<e style="counter-increment: element other -2;"><span>Element / 21. -2</span><span></span></e>
<e length="100" style="counter-increment: element other -2;"><span>Element / 22. -4</span><span>Test</span><span> | Length: 100 [xxii:V]</span></e>
<f style="counter-increment: other;"><span>-3</span><span></span></f>
<f style="counter-increment: other;"><span>-2</span><span></span></f>
</body></html>
